lib/interpose.c

```diff
@@ -730,8 +801,9 @@ realloc(void *ptr, size_t size)
 {
   void *result;
 
-  if (real_realloc == NULL)
-    __ulp_asunsafe_begin();
+  if (real_realloc == NULL) {
+    return __libc_realloc(ptr, size);
+  }
```

Comment:
For my understanding: you aren't calling ulp_unsafe_begin on demand anymore (here and in the rest of the changed functions).  How is that working, IOW: where is it called now?

Reply:
`__ulp_asunsafe_begin` is a ctor, that means it is called when libpulp is loaded. It is still being called on every interposed libdl symbol.

Reply:
Ah, right.  This all seems fine then.